File: example-cs/Puzzle.cs
internal class Puzzle : AocPuzzle<int[], int>
{
    protected override int[] ParseInput(string[] lines) => lines.Select(int.Parse).ToArray();

    protected override int RunPartOne() => Input.Sum();

    protected override int RunPartTwo() => Input.Aggregate((prev, cur) => prev * cur);
}

File: day01/Puzzle.fs
module Puzzle

let private getCalories (elves: int[][]) =
  elves |> Array.map Array.sum


let parseInput lines =
  (lines |> String.concat "\n").Split("\n\n")
    |> Array.map (fun elf -> elf.Split("\n") |> Array.map int)

let runPartOne input =
  input
  |> getCalories
  |> Array.max

let runPartTwo input =
  let cals =
    input
    |> getCalories
    |> Array.sortDescending

  cals[0] + cals[1] + cals[2]
